subsys/net/lib/nrf_cloud/coap/src/nrf_cloud_coap_transport.c

```diff
@@ -113,9 +118,45 @@ static const char *fmt_name(enum coap_content_format fmt)
 }
 #endif /* CONFIG_NRF_CLOUD_COAP_LOG_LEVEL_DBG */
 
+static struct cc_xfer_data *xfer_ctx_take(void)
+{
+	for (int i = 0; i < ARRAY_SIZE(xfer_ctx_pool); i++) {
+		if (!atomic_test_and_set_bit(&xfer_ctx_pool[i].used, 0)) {
+			return &xfer_ctx_pool[i];
+		}
+	}
+	return NULL;
+}
+
+static void xfer_ctx_free(struct cc_xfer_data *ctx)
+{
+	if (ctx) {
+		atomic_clear_bit(&ctx->used, 0);
+	}
+}
+
+static struct cc_xfer_data *xfer_data_init(struct nrf_cloud_coap_client *cc,
+					   coap_client_response_cb_t cb,
+					   void *user,
+					   struct k_sem *sem)
+{
+	struct cc_xfer_data *xfer = xfer_ctx_take();
+
+	if (!xfer) {
+		LOG_ERR("No free xfer_ctx");
```

Comment:
```suggestion
		LOG_ERR("Maximum number of CoAP transfers are already in progress");
```
or something like that